### PR TITLE
Amorphic client bootstrap - update on Lysandros' mr

### PR DESCRIPTION
**Pull request by sshankar19**
From lysandros' mr

> The scope of this PR is to set the groundwork for eliminating the dependency of amorphic to bindster.
In current applications we may not need to use Bindster anymore.
However, the amorphic client was being initialized inside the bindster-amorphic.js which is inside the amorphic-bindster module. Thus, in those apps we're importing this file within index.html and using a hacky script to override the bindster methods:

> <script> function Bindster() {} Bindster.prototype.start = function () {} Bindster.prototype.setModel = function() {} Bindster.prototype.setController = function() {} </script>
> I created a new file called clientInit.js at the root of the amorphic project.
Importing just this file is enough to initialize the amorphic client (the establishClientSession is invoked at the very end)
> I went through some refactoring of that code as well. All of the bindster related code is removed and another variable is used instead (firstTimeLoad). Used consts were posssible, changed a log entry and pulled all the function arguments outside.
> This change was tested in current applications (their index.html was modified to import the new file and the hacky script was removed).
> ** A future PR could also stop serving bindster through amorphic all together. For this to happen, anything that uses amorphic and relies on that dependency to also use bindster should import bindster 'itself'.

I've retested the change and also updated the package.json and package-lock.json

On top of that these are the new changes you will have to make in your app in order to use this approach.

Whereas before if you had a non-bindster application using a different UI framework (such as angular) you might have to add these script tags to you index.html
`  <script>
        function Bindster() {}
        Bindster.prototype.start = function () {}
        Bindster.prototype.setModel = function() {}
        Bindster.prototype.setController = function() {}
    </script>`

and 
`<script src="/bindster/bindster-amorphic.js"></script>`

Now you can delete the both blocks. Instead of the latter block which includes bindster-amorphic, you should replace that with this instead
`<script src="/amorphic/clientInit.js"></script>`

### CODE REVIEW

File: clientInit.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+var controller = typeof(controller) === 'undefined' ? null : controller;
+var firstTimeLoad = typeof(firstTimeLoad) === 'undefined';
+
+// Establish a new session whether first time or because of expiry / server restart
+function __bindController (newController, sessionExpiration) {
+    if (controller && typeof(controller.shutdown) === 'function') {
+        controller.shutdown();
+    }
+    controller = newController;
+    if (firstTimeLoad) {
```
Comment (CSMastermind) [inline clientInit.js]: Do you need this `if`?  Can't you always just set it to false?

Comment (sshankar19) [reply clientInit.js]: I'm actually not sure why @lysandros even made this variable firstTimeLoad. It doesn't actually do _anything_. It's just initiliazed to true in this file and then switched to false in bindController.

Comment (lysandros) [reply clientInit.js]: It made sense sometime ago when controller.refresh() was the else {} part. 
We can remove the variable and the check.

**Comment by CSMastermind** [comment]
👍 Looks good to me

**Review by nikmash — APPROVED** [review]
Looks good

**Pull request merged**